### ISSUE TITLE
Setting partition rules in gRPC table creation request

### ISSUE THREAD

**Issue by MichaelScofield** [GreptimeTeam/greptimedb#1709]
### What type of enhancement is this?

User experience

### What does the enhancement do?

Now the partition rules cannot be represented in our gRPC table creation request: `CreateTableExpr`. Which means a distributed table can only be created via SQL if it has to explictly state partition rules.

**Comment by NiwakaDev** [comment]
I'd like to work on this issue.

**Comment by MichaelScofield** [comment]
@NiwakaDev welcome!

**Comment by NiwakaDev** [comment]
Do I need to address this issue after refactoring `region-server`?

**Comment by MichaelScofield** [comment]
@NiwakaDev Yes, the "table" concept still exist. When the Datanode is done turning into a region server, it's just the grpc  between Frontend and Datanode that don't have the table creation requests, but the grpc between Frontend and the client do have.

**Comment by NiwakaDev** [comment]
@MichaelScofield 

It seems like Datanode is a region server now, so I guess that we can address this issue.

**Comment by MichaelScofield** [comment]
@NiwakaDev I'm afraid we can't. We now have two set of grpc interfaces, one is for Frontend communicating with Datanode, and the other is for user to Frontend. The former one is indeed not need to carry any partition rules, but the later one does.

**Comment by NiwakaDev** [comment]
>We now have two set of grpc interfaces, one is for Frontend communicating with Datanode, and the other is for user to Frontend. The former one is indeed not need to carry any partition rules, but the later one does.

The former one is `RegionRequest` and the later one is `CreateTableExpr`? 

If so, as far as I understand, the only later one is related to this issue, not the former one. So, if the former one doesn't interfere with the later one, I guess that we can address this issue. Why can't we address this issue?

**Comment by MichaelScofield** [comment]
> > We now have two set of grpc interfaces, one is for Frontend communicating with Datanode, and the other is for user to Frontend. The former one is indeed not need to carry any partition rules, but the later one does.
> 
> The former one is `RegionRequest` and the later one is `CreateTableExpr`?

yes

> If so, as far as I understand, the only later one is related to this issue, not the former one. I guess that the former one doesn't interfere with the later one. Why can't we address this issue?

We can set the partition rules in `CreateTableExpr`, but not the `RegionRequest`. Sorry about the misunderstanding.

**Comment by fengjiachun** [comment]
Can we close it now? @MichaelScofield

**Comment by MichaelScofield** [comment]
@fengjiachun No, we still lack of partition rules in grpc create table request:

```protobuf
message CreateTableExpr {
  string catalog_name = 1;
  string schema_name = 2;
  string table_name = 3;
  string desc = 4;
  repeated ColumnDef column_defs = 5;
  string time_index = 6;
  repeated string primary_keys = 7;
  bool create_if_not_exists = 8;
  map<string, string> table_options = 9;
  TableId table_id = 10;
  string engine = 12;
}
```
